From dab5e75a1a0a72ee3c97814c736dad6c44087c32 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 13 Jan 2021 14:30:45 +0530
Subject: [PATCH] wifi: Disable capabilities not yet supported

As E2E support is not yet implemented for following capabilities,
removing them.
DEBUG_RING_BUFFER_VENDOR_DATA, DEBUG_HOST_WAKE_REASON_STATS,
and APF.

Tracked-On: OAM-95699
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 wifi/1.4/default/hidl_struct_util.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/wifi/1.4/default/hidl_struct_util.cpp b/wifi/1.4/default/hidl_struct_util.cpp
index fd1d5b112..081257113 100644
--- a/wifi/1.4/default/hidl_struct_util.cpp
+++ b/wifi/1.4/default/hidl_struct_util.cpp
@@ -157,8 +157,8 @@ bool convertLegacyFeaturesToHidlChipCapabilities(
 
     // There are no flags for these 3 in the legacy feature set. Adding them to
     // the set because all the current devices support it.
-    *hidl_caps |= HidlChipCaps::DEBUG_RING_BUFFER_VENDOR_DATA;
-    *hidl_caps |= HidlChipCaps::DEBUG_HOST_WAKE_REASON_STATS;
+    //*hidl_caps |= HidlChipCaps::DEBUG_RING_BUFFER_VENDOR_DATA;
+    //*hidl_caps |= HidlChipCaps::DEBUG_HOST_WAKE_REASON_STATS;
     *hidl_caps |= HidlChipCaps::DEBUG_ERROR_ALERTS;
     return true;
 }
@@ -379,7 +379,7 @@ bool convertLegacyFeaturesToHidlStaCapabilities(
         return false;
     }
     *hidl_caps = {};
-    using HidlStaIfaceCaps = IWifiStaIface::StaIfaceCapabilityMask;
+    //using HidlStaIfaceCaps = IWifiStaIface::StaIfaceCapabilityMask;
     for (const auto feature : {legacy_hal::WIFI_LOGGER_PACKET_FATE_SUPPORTED}) {
         if (feature & legacy_logger_feature_set) {
             *hidl_caps |=
@@ -399,7 +399,7 @@ bool convertLegacyFeaturesToHidlStaCapabilities(
     }
     // There is no flag for this one in the legacy feature set. Adding it to the
     // set because all the current devices support it.
-    *hidl_caps |= HidlStaIfaceCaps::APF;
+    //*hidl_caps |= HidlStaIfaceCaps::APF;
     return true;
 }
 
-- 
2.17.1

